hub/server.go

```diff
@@ -179,3 +187,24 @@ func welcomeHandler(w http.ResponseWriter, r *http.Request) {
 <title>Mercure Hub</title>
 <h1>Welcome to <a href="https://mercure.rocks">Mercure</a>!</h1>`)
 }
+
+func basicAuthMiddleware(expectedLogin, expectedPassword string) func(next http.Handler) http.Handler {
+	return func(next http.Handler) http.Handler {
+		return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+			login, password, ok := r.BasicAuth()
+			if !ok {
```

Comment:
Both `if` blocks can be merged:

```suggestion
			if if !ok || login != expectedLogin || password != expectedPassword {
```